jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java

```diff
@@ -128,6 +132,32 @@ public Server(@Name("threadpool") ThreadPool pool)
         addBean(_threadPool);
         setServer(this);
         addBean(FileSystemPool.INSTANCE, false);
+        setDefaultHandler(new DefaultHandler());
```

Comment:
Should we set a `DefaultHandler` by default?  That is a little behavior change, but the clue is in the name!

Reply:
You could, but then if someone wants to configure the `DefaultHandler` they have to `new` it again in the xml or code. Instead you could check in `doStart` if there's no `_defaultHandler` configured, then instantiate our `DefaultHandler`.

Reply:
Note that we don't strictly need a default handler as we will generate a 404 fine without one.   We only really need it for the convenience of getting a context list on a root 404, plus the marketing of serving fav.icon.

Perhaps we just get rid of the whole default handler concept, as it is not really needed?

Reply:
Feels weird that we serve `jetty-dir.css` from `ResourceHandler`, but `favicon.ico` from the `DefaultHandler`, given the files are actually in the same place.

If we move out the serving of `favicon.ico`, what remains is the small HTML to present the list of contexts, which is useful, and people may want to replace it, so I think the concept of a `DefaultHandler` should remain.

Reply:
Since we set a `DefaultHandler` in our `jetty.xml`, like we always have, I have made the embedded server no have a default handler, unless one is set. That is the same behavior as before and a simple raw 404 will be produced by the server.

> Feels weird that we serve jetty-dir.css from ResourceHandler, but favicon.ico from the DefaultHandler, given the files are actually in the same place.

The two are different. We want jetty-dir.css to appear in every directory in the content tree, whilst favicon.ico is only in the top level, outside of any context.
Furthermore, jetty-dir.css is necessary for a feature we offer: listing directories, whilst favicon.ico` is a marketing stunt of dubious value.
We could definitely remove favicon.ico and nobody would care/complain and many would be happy to not have to turn it off!, but now that it is not set by default (unless using our module which is easily configured), I'm OK keeping it.